crafters/image/ImageTorchTransformation/__init__.py

```diff
@@ -0,0 +1,165 @@
+
+        The torch equivalent for this would be recursively apply `Compose` on each image.
+
+        .. code-block:: python
+            import torchvision.transforms as T
+
+            transforms = T.Compose(
+                [
+                    T.ToTensor(),
+                    T.CenterCrop(300),
+                    T.Resize((244, 244)),
+                    T.RandomVerticalFlip(p=1.0),
+                    T.Normalize(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)),
+                ]
+            )
+
+            crafted_img = transforms(image_batch[0]).cpu().numpy()
+            crafted_img = np.transpose(crafted_img, (1, 2, 0))
+
+
+    :param transforms: List of transforms that are applied sequentially
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    def __init__(
+        self,
+        transforms: List[Union[str, Dict[str, Dict]]],
+        *args,
+        **kwargs,
+    ):
+        """Set constructor."""
+        super().__init__(*args, **kwargs)
+        if not isinstance(transforms, list):
+            self.logger.error("The `transforms` argument has to be a list.")
+            raise ValueError
+
+        self.transforms_specification = transforms
+
+    def post_init(self):
+        from torchvision import transforms as T
+
+        transforms_list = []
+        for transform in self.transforms_specification:
+            if isinstance(transform, str):
+                tr_name, tr_kwargs = transform, {}
+            elif isinstance(transform, dict):
+                tr_name, tr_kwargs = next(iter(transform.items()))
+            else:
+                self.logger.error(
+                    "Each element of the `transform` has to be either a dict or  a string"
+                )
+                raise ValueError
+
+            try:
+                tr_class = getattr(T, tr_name)
+            except AttributeError:
+                self.logger.error(
+                    f"The transform class `{tr_name}` does not seem to exist,"
+                    "check that you have not made a typo."
+                )
+                raise ValueError
+
+            if "p" in tr_class.__init__.__code__.co_varnames:
+                # does not handle inherently random transform like `RandomErasing`
+                tr_kwargs.update({"p": 1.0})
+
+            try:
+                transform_instance = tr_class(**tr_kwargs)
+            except:
+                self.logger.error(
+                    f"Error instantiating torchvision transforms class `{tr_name}`,"
+                    " check if all kwargs are valid and if transform is valid for tensors."
+                )
+                raise ValueError
+
+            transforms_list.append(transform_instance)
+        self.transforms = T.Compose(transforms_list)
+
+    def craft(
+        self, blob: Union[List["np.ndarray"], "np.ndarray"], *args, **kwargs
+    ) -> List[Dict]:
```

Comment:
blob is np.array, not need to handle lost

Reply:
ah no, my bad

Reply:
Ya, since it accepts batch and does not have @single

Reply:
yes, actually it will only be List of numpy arrays.

But I am not sure u can stack them

Reply:
Makes sense. But for reason, stacking was an important test in albumentationsCrafter

Reply:
but I am wondering now if this stack will work when images of different shapes are passed.

Thats why I insist on testing with a batch of images of different dimensions

Reply:
Ah! I understand what you mean now. All the 'batch' test that I did was with the same image (shape) even though there were 2-5.

What you suggest is [small_img, big_img] etc. That's a good point, didn't think in that direction.
For any dataset, the images would ideally be of the same dimensions. But definitely, there could be someone who pools images expecting to handle them here with transforms. 
In that scenario, you cant batch anyway, not with this implementation but in general, as it the tensor should be BCHW

One should do use ImageCropper first to make them the same shape and then use this crafter to do other transforms faster in batch.

Reply:
Let's put that clearly in the `README` that input needs to be of the same shape

Reply:
Sure, I added a clear note now.